upload/system/Login/Steam.php

```diff
@@ -238,60 +248,52 @@ public function canChange($type, \IPS\Member $member)
 		return false;
 	}
 
+	/**
+	 * Gets a url param and sanitizes it for openid
+	 *
+	 * @param    string    $param  url param
+	 *
+	 * @return    string
+	 */
+	private function get($param)
+	{
+		return str_replace(' ', '+', trim(urldecode(\IPS\Request::i()->{$param})));
```

Comment:
In my experience, it's best to the parameter without it having gone through IPS parameter sanitization. Tricks like replacing spaces with +'s aren't always reliable as I've seen them fail in the past.

Reply:
Unless there's a way to tell when ipb force urlencodes it that's the best I can think of.

Reply:
Just use `$_GET`, as the previous code did, because IPS doesn't modify the global variable. ;) It's safe to use it directly.

Reply:
Alright, this is the only thing I'm against pulling in. If you change this to 

```php
        $value = isset($_GET[$param]) ? $_GET[$param] : null;

        if ($value === null) {
			return null;
        }

		return trim(urldecode($value));
```

Then 👍

If only IPS's min requirement was PHP 7.0 and not 5.6 then it could be

```php
        if ($value = $_GET[$param] ?? false) {
			return null;
        }

		return trim(urldecode($value));
```

Reply:
well... php 7.0+ is their recommended. 5.6 and 7.0 are EOL, 7.1 is almost there, 7.2 being current. why not start supporting at least 7.0 only.  if someones host still has them on 5.6 or lower, they need to re-evaluate their webhosting.

Reply:
It's not worth the aggravation when it's easy enough to support. 

FYI PHP 5.6 &  7.0 become unsupported versions this year [source](https://secure.php.net/supported-versions.php), which really leaves 7.1 / 7.2 as "Officially support by PHP Maintainer". 

Some additional FYI's is that PHP 5.6, 7.0 (and possibly older ones) are still going to be around for a while as vendors such as Canonical and Red Hat will maintain support for LTS distributions that contain PHP binaries in their official repos that they offer LTS support on.